File: Backend/productManagement/src/main/java/com/management/productManagement/service/ProductService.java
package com.management.productManagement.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.management.productManagement.entity.Product;
import com.management.productManagement.repo.productrepo;

@Service
public class ProductService {

	@Autowired
	private productrepo productRepo;


	public Product saveProduct(Product product) {

		return productRepo.save(product);
	}


	public List<Product> getAllProduct() {
		return productRepo.findAll();
	}

	public Product getProductById(Integer id) {
		return productRepo.findById(id).get();
	}


	public String deleteProduct(Integer id) {
		Product product = productRepo.findById(id).get();

		if (product != null) {
			productRepo.delete(product);
			return "Product Delete Sucessfully";
		}

		return "Something wrong on server";
	}


	public Product editProduct(Product p, Integer id) {

		Product oldProduct = productRepo.findById(id).get();

		oldProduct.setProductName(p.getProductName());
		oldProduct.setDescription(p.getDescription());
		oldProduct.setPrice(p.getPrice());
		oldProduct.setStatus(p.getStatus());

		return productRepo.save(oldProduct);
	}

}

File: Frontend/product-frontend/src/service/productservice.js
import axios from "axios";

const API_URL="http://localhost:8080/product";

class ProductService {

   saveProduct(product) {  
     return axios.post(API_URL + "/saveProduct", product);
    }

    getAllProducts() {
        return axios.get(API_URL + "/getAllProducts");
    }

    getProductById(id) {
        return axios.get(API_URL + "/" + id);
    }

    deleteProduct(id) {
        return axios.delete(API_URL + "/deleteProduct/" + id);
    }

    editProduct(product) {
        return axios.post(API_URL + "/editProduct/" + product.id, product);
    }
}

export default new ProductService;